gadget/layout.go

```diff
@@ -189,6 +192,69 @@ func LayoutVolumePartially(volume *Volume, constraints LayoutConstraints) (*Part
 	return vol, nil
 }
 
+func resolveOne(gadgetRootDir, kernelRootDir string, kernelInfo *KernelInfo, pathOrRef string) (string, error) {
+	// content may refer to "$kernel:<name>/<content>"
+	if strings.HasPrefix(pathOrRef, "$kernel:") {
+		kernelRef := strings.SplitN(pathOrRef, ":", 2)[1]
+		l := strings.SplitN(kernelRef, "/", 2)
```

Comment:
Is it always `$kernel:<asset>/<content>`, or can we use `$kernel:<asset>` to mean all content too?

Reply:
I don't think this was speced in detail, for now I would start with $kernel:assets/<content> only

Reply:
I updated this now to use a regexp and it is now stricter.